### DISCUSSION TITLE
A common  error format is needed for the  evaluate function

### DISCUSSION THREAD

**Discussion by sparkidev0x1D** [josdejong/mathjs#3060] [Ideas]
Currently, the errors are being thrown  in different structures one may be in an array the other in an object model 

It's more suitable to have a common data structure for the errors being thrown 

**error for 
'2C + '**
```
'SyntaxError: Unexpected end of expression (char 6)'
[
  'end of expression',
  index: 24,
  input: 'SyntaxError: Unexpected end of expression (char 6)',
  groups: undefined
]

```

some may be in array in nature, the other may have a Data Object  inside it 

error for missing a binding parameter on the expression

```
'TypeError: Unexpected type of argument in function addScalar (expected: Unit, actual: number, index: 1)'
{
  category: 'wrongType',
  fn: 'addScalar',
  index: 1,
  actual: 'number',
  expected: [ 'Unit' ]
}
```

**Comment by josdejong** [comment]
Thanks for your suggestion. I guess we can improve there. Currently there are a couple of errors like `SyntaxError`, `DimensionError`, `ArgumentsError` which each attach some relevant properties like `.char` or `.expected`. I don't think this is currently documented. 

Sharukh do you have any idea on how to unify this? Or would it be enough to document these properties?

**Comment by sparkidev0x1D** [comment]
I think we can  implement a common error handler for the Evaluation side as of initial stage  and then it could be extended to other modules, Hope it makes sense 
Currently, I am  a bit busy with other projects hope I could integrate it by the end of March

**Comment by josdejong** [comment]
I'm not sure what you mean with an error handler, can you elaborate or show some pseudo code to explain what you mean? Right now Errors are thrown when there is an issue, and you can catch them with try/catch.

**Comment by gwhitney** [comment]
While consistency in error formats would be very good, there is not yet a clear actionable proposal here, so moving this to Discussions. See also #3055 which touches on error formats.
